greenpithumb/greenpithumb.py

```diff
@@ -29,35 +29,135 @@
-                                    moisture_threshold, pump_amount, pump_timer)
+        clock.Clock())
+    return temperature_sensor.TemperatureSensor(
+        local_dht11), humidity_sensor.HumiditySensor(local_dht11),
+
+
+def make_soil_moisture_sensor(adc, raspberry_pi_io, wiring_config):
+    return soil_moisture_sensor.SoilMoistureSensor(
+        adc, raspberry_pi_io, wiring_config.adc_channels.soil_moisture_sensor,
+        wiring_config.gpio_pins.soil_moisture)
+
+
+def make_light_sensor(adc, wiring_config):
+    return light_sensor.LightSensor(adc,
+                                    wiring_config.adc_channels.light_sensor)
+
+
+def make_camera_manager(rotation, image_path, light_sensor):
+    """Creates a camera manager instance.
+
+    Args:
+        rotation: The amount (in whole degrees) to rotate the camera image.
+        image_path: The directory in which to save images.
+        light_sensor: A light sensor instance.
+
+    Returns:
+        A CameraManager instance with the given settings.
+    """
+    camera = picamera.PiCamera(resolution=picamera.PiCamera.MAX_RESOLUTION)
+    camera.rotation = rotation
+    return camera_manager.CameraManager(image_path,
+                                        clock.Clock(), camera, light_sensor)
+
+
+def make_pump_manager(moisture_threshold, sleep_windows, raspberry_pi_io,
+                      wiring_config, pump_amount, pump_interval):
+    """Creates a pump manager instance.
+
+    Args:
+        moisture_threshold: The minimum water level before the pump turns on.
+        sleep_windows: Sleep windows during which pump will not turn on.
+        raspberry_pi_io: pi_io instance for the GreenPiThumb.
+        wiring_config: Wiring configuration for the GreenPiThumb.
+        pump_amount: Amount (in mL) to pump on each run of the pump.
+        pump_interval: Maximum amount of time between pump runs.
+
+    Returns:
+        A PumpManager instance with the given settings.
+    """
+    water_pump = pump.Pump(raspberry_pi_io,
+                           clock.Clock(), wiring_config.gpio_pins.pump)
+    pump_scheduler = pump.PumpScheduler(clock.LocalClock(), sleep_windows)
+    pump_timer = clock.Timer(clock.Clock(), pump_interval)
+    return pump.PumpManager(water_pump, pump_scheduler, moisture_threshold,
+                            pump_amount, pump_timer)
+
+
+def make_sensor_pollers(poll_interval, photo_interval, record_queue,
+                        temperature_sensor, humidity_sensor,
+                        soil_moisture_sensor, light_sensor, camera_manager,
+                        pump_manager):
+    """Creates a poller for each GreenPiThumb sensor.
+
+    Args:
+        poll_interval: The frequency at which to poll non-camera sensors.
+        photo_interval: The frequency at which to capture photos.
+        record_queue: Queue on which to put sensor reading records.
+        temperature_sensor: Sensor for recording temperature.
```

Comment:
(optional) Maybe more accurate to say 'Sensor for capturing temperature'? Ditto for the other sensors.